### ISSUE TITLE
exclude notes from word count

### ISSUE THREAD

**Issue by queuingforsarnies** [olivierkes/manuskript#451]
as discussed here: https://github.com/olivierkes/manuskript/issues/247 a notes file can be excluded from compiling -- but is there a way to exclude it from the total word count that displays in the bottom right of the window?

**Comment by gedakc** [comment]
Currently the word count indicates the total for the items displayed in the center view.  For an example see [this comment in issue 245](https://github.com/olivierkes/manuskript/issues/245#issuecomment-348961192).  This includes all text in the displayed section regardless of the **Compile** settings.

Following is an example of the word count for a whole document (includes compile disabled text "Appendix B"):

![manuskript-whole-document-word-count](https://user-images.githubusercontent.com/10405019/50611878-4316e280-0e95-11e9-80c1-ff8513bcb572.png)

I'll leave this issue open as a future feature request.

**Comment by kakaroto** [comment]
It should be easy to have a folder "Book" and a folder "Notes", and move all your chapters into the "Book" as subfolders, then you'd check the total words written by selecting "Book" instead of "Root'. It will be annoying though to move everything to a new folder, but you only have to do it once and it shouldn't be too bad unless you have hundreds of chapters.
I also like this feature request, but there are easy workarounds.

**Comment by mxamber** [comment]
Went looking for something like this and would like to second the notion of implementing this as a feature in the future. Thanks for the suggestion of moving it into different folders tho, that's enough of a fix for me for now.

**Comment by mxamber** [comment]
Just went to look whether this was already an open issue before posting, and turns out it is and has been for three years. Are there any plans to add this in the future? I just thought under Settings -> Views ->  Tree -> Word Counter would be an ideal place to put a checkbox like "exclude uncompiled items from word count" or something like that. Due to the nature of my current project (author's notes per chapter), putting the notes in a separate folder is not an option either.

**Comment by TheJackiMonster** [comment]
I assume this could be implemented like this but probably nobody picked it up to implement it because there was mentioned a workaround. I can definitely keep this issue in mind while refactoring the GUI.

**Comment by TheJackiMonster** [comment]
I have looked into the problem again. So my conclusion is that it would make sense to add a checkbox to the dialog of creating labels. So you could activate/deactivate the functionality of counting words/chars for the specific part of the project as child of others (like notes for example to chapters).

The problem why we shouldn't just put a static filter inside the code to prevent notes from being counted is that notes are just an arbitrary example for a useful label. This label could be renamed though or removed completely. So it should work independent of a labels name with a new attribute.

I will put this issue on the list for future releases since this would require more detailed work to integrate. The next release (0.13.0) will mostly focus on bug fixes for now.

**Comment by mxamber** [comment]
What about tying it to the "Compile" checkbox instead? Excluding every text that won't be compiled from being counted?

**Comment by TheJackiMonster** [comment]
@mxamber I agree that this would make sense but it's not ideal since the compile flag is implemented as filter during the process of compilation/export. So while it could work in practice it would be kind of awkward to go to "Compile", find the filters, set them up properly and go back to the text editor to look at the word count. Not intuitive either.

I'm not 100% sure if the compile filters are bound to each project individually either but that's my assumption as well. However labels and their properties (for example color and name) should persist between projects. So it would be easier to flag parts of your project to exclude them from counting.

I would also suggest that we might even add the compile option to the labels as well instead of making it an advanced part of the compile dialog. I would think that this could help overall usability.

**Comment by Anne501R** [comment]
Hi, first much thanks for this great software and the exisiting discussion. It would also ease my working with Manuskript, if labels would have additional functionality / configurations in the label settings for whether or not to ...

- count words
- add it to frequency analyzer (or this might be included in word count)
- add it to story line overview
- compile

sidenote / further thoughts: If further configuration / functionality is added to labels, I wonder, whether a label would still be a 'label'. Maybe it is more like an object / file type. And in this way of thinking the all-in-all desired feature might be more like a hidden / note / research file (no word count, no fequency analyzer, no story line, no compilation) in addition to the current text file. - But that is open for discussion.

**Comment by mxamber** [comment]
> sidenote / further thoughts: If further configuration / functionality is added to labels, I wonder, whether a label would still be a 'label'. Maybe it is more like an object / file type. And in this way of thinking the all-in-all desired feature might be more like a hidden / note / research file (no word count, no fequency analyzer, no story line, no compilation) in addition to the current text file. - But that is open for discussion.

Seconded! For me, it's stuff like author's notes (fanfic stuff) and discarded drafts or alternate versions of chapters I'd like to keep in the project for convenience but exclude from compile and statistics.
